python/tk_framework_desktopserver/shotgun/api_v2.py

```diff
@@ -221,13 +221,23 @@ def execute_action(self, data):
             "execute_command.py"
         )
 
-        # We'll need the Python executable when we shell out. We can't
-        # rely on sys.executable, because that's going to be the Desktop
-        # exe on Windows. We'll pull from the site config's interpreter
-        # cfg file.
-        python_exe = sgtk.get_python_interpreter_for_config(
-            self._engine.sgtk.pipeline_configuration.get_path(),
-        )
+        # We'll need the Python executable when we shell out. We need to make
+        # sure it's what's in the config's interpreter config file. To do that,
+        # we can get the config's descriptor object from the manager and ask
+        # it for the path. By this point, all of the config data has been cached,
+        # because it will have been looked up as part of the get_actions method
+        # when the menu asked for actions, so getting the manager and all of the
+        # config data will be essentially free.
+        manager = self._get_toolkit_manager()
```

Comment:
I notice the lock in other PRs but didn't give much thought to it. Have you considered creating one on the fly instead so each thread has its own? This would remove the need for the lock, right?